gatewayconfig/bluetooth/characteristics/firmware_revision_characteristic.py

```diff
@@ -12,12 +13,8 @@ def __init__(self, service):
                 ["read"], service)
 
     def ReadValue(self, options):
-        logging.debug('Read Firmware')
-
-        val = os.getenv('FIRMWARE_VERSION')
-        value = []
-
-        for c in val:
-            value.append(dbus.Byte(c.encode()))
-
-        return value
\ No newline at end of file
+        logger.debug('Read Firmware')
+        # Intentionally reading the env variable each time in case it changes over
+        # the application's lifetime
+        firmware_version = os.getenv('FIRMWARE_VERSION')
```

Comment:
Actually I don't think balena can upgrade versions without restarting the app, can it?

Reply:
I think it depends. But in theory if you only updated the miner container then it could. 

However, wouldn't updating the firmware version in the docker-compose.yml cause the config container to restart anyway? 
https://github.com/NebraLtd/helium-miner-software/blob/0158481bc872097b25ee644b5df83a5bd7215783/docker-compose.yml#L8

Reply:
Any changes to an environment variable in `docker-compose.yml` will cause the container to restart.

Reply:
So we don't have to read from the ENV every time and can do it once at the start of the app. We can make that optimization in the future.